### PR TITLE
add featuregate for authorization

### PR DESCRIPTION
**Pull request by vincentgoat**
Signed-off-by: vincentgoat <linguohui1@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Addition of PR: https://github.com/kubeedge/kubeedge/pull/4104
add featuregate for authorization

Featuregate of `requireAuthorization` default is false as it is in the alpha stage, follow below to enable this gate.
edgecore.yaml:
```
apiVersion: edgecore.config.kubeedge.io/v1alpha1
database:
  aliasName: xxx
  ...
featureGates:
  requireAuthorization: true
kind: EdgeCore
modules:
  ...
```
cloudcore.yaml:
```
apiVersion: cloudcore.config.kubeedge.io/v1alpha1
commonConfig:
  tunnelPort: xxx
kind: CloudCore
featureGates:
  requireAuthorization: true
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -19,13 +19,14 @@ func init() {
...
+	// alpha: v1.12
+	RequireAuthorization featuregate.Feature = "requireAuthorization"
 )
 
 // defaultFeatureGates consists of all known Kubeedge-specific feature keys.
 // To add a new feature, define a key for it above and add it here. The features will be
 // available throughout Kubeedge binaries.
 var defaultFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
-	ProcessStatusSync: {Default: false, PreRelease: featuregate.Alpha},
+	RequireAuthorization: {Default: true, PreRelease: featuregate.Alpha},
```
Comment (wackxu) [inline pkg/features/features.go:31]: this is strange, it is a alpha feature but the default value is true

File: pkg/features/features.go [hunk 2]
```diff
@@ -19,13 +19,14 @@ func init() {
 }
 
 const (
-	// ProcessStatusSync supports synchronization between process message channel statuses
-	ProcessStatusSync featuregate.Feature = "ProcessStatusSync"
+	// RequireAuthorization Supports application access authorization from edge sides
+	// alpha: v1.12
+	RequireAuthorization featuregate.Feature = "requireAuthorization"
```
Comment (wackxu) [inline pkg/features/features.go:24]:
```suggestion
	// owner: @vincentgoat
	// alpha: v1.12

	// RequireAuthorization Supports application access authorization from edge sides
```

Comment (wackxu) [reply pkg/features/features.go:24]: the comment is less accurate，please add more detail

File: pkg/features/features.go (hunk 1 above)
Comment (fisherxu) [reply pkg/features/features.go:31]: Agreed, default value should be false? If users want to use it, just set it in config file

Comment (vincentgoat) [reply pkg/features/features.go]: done

File: pkg/features/features.go (hunk 2 above)
Comment (vincentgoat) [reply pkg/features/features.go:28]: done

**Review by fisherxu — COMMENTED** [review]
/lgtm

**Pull request merged**